addons/mail/models/mail_link_preview.py

```diff
@@ -0,0 +1,141 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, models, fields
+
+from datetime import datetime
+from dateutil.relativedelta import relativedelta
+from lxml import html, etree
+from urllib.parse import urlparse
+import requests
+
+
+class LinkPreview(models.Model):
+    _name = 'mail.link.preview'
+    _description = "Store link preview data"
+
+    message_id = fields.Many2one('mail.message', string='Message', index=True, ondelete='cascade', required=True)
+    guest_id = fields.Many2one('mail.guest', string="Guest")
+    source_url = fields.Char('url', required=True)
+    og_type = fields.Char('type')
+    og_title = fields.Char('title')
+    og_image = fields.Char('image')
+    og_description = fields.Text('description')
+    og_mimetype = fields.Char('mimetype')
+    image_mimetype = fields.Char('image_mimetype')
+
+    # Some website are blocking non browser user agent. We make ourself like
+    # firefox to avoid issue.
+    request_headers = {
+        'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; rv:91.0) Gecko/20100101 Firefox/91.0'
+    }
+
+    @api.model
+    def _create_link_preview(self, url, message_id):
+        if self._is_domain_throttled(url):
+            return False
+        link_preview_data = self._get_link_preview_from_url(url)
+        if link_preview_data:
+            if self.env.user._is_public() and 'guest' in self.env.context:
+                link_preview_data['guest_id'] = self.env.context['guest'].id
+            link_preview_data['message_id'] = message_id
+            return self.sudo().create(link_preview_data)
+        return False
+
+    def _delete_and_notify(self):
+        notifications = []
+        for link_preview in self:
+            if link_preview.message_id.model == 'mail.channel' and link_preview.message_id.res_id:
+                target = self.env['mail.channel'].browse(link_preview.message_id.res_id)
+            else:
+                target = self.env.user.partner_id
```

Comment:
Alright, but then it might be useful to check the other _delete_and_notify method from attachment:

https://github.com/odoo-dev/odoo/blob/master-mail-implement_opengraph_attachment-did/addons/mail/models/ir_attachment.py#L44